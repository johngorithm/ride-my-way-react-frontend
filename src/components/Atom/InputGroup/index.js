import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import InputField from 'components/Atom/InputField';

const InputGroup = ({ classes, error, label, ...props}) => (
  <Fragment >
    <div className={classes}>
      <p className="smaller">
        {label} {error ? <span className="error-message">{error}</span> : null } 
      </p>
      <InputField {...props} />
    </div>
  </Fragment>
);

InputGroup.propTypes = {
  label: PropTypes.string,
  error: PropTypes.string
};

export default InputGroup;