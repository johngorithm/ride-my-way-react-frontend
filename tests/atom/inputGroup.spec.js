import React from 'react';
import { shallow } from 'enzyme';

import InputGroup from 'components/Atom/InputGroup';

const props = {
  lable: 'Password',
  error: 'this field is required',
  name: 'password',
  type: 'text',
  value: 'fd89ya8sdf',
  onChange: jest.fn()
}

describe('Button Tests', () => {
  test('it renders successfully', () => {
    const wrapper = shallow(<InputGroup {...props} />);
    expect(wrapper.exists()).toBe(true);
  })

  test('it matches snapshot', () => {
    const wrapper = shallow(<InputGroup {...props}/>);
    expect(wrapper).toMatchSnapshot();
  })

  test('it does not render error message when error is undefined', () => {
    const attr = { lable: 'Name', error: undefined }
    const wrapper = shallow(<InputGroup { ...attr } />);
    expect(wrapper).toMatchSnapshot();
  })
})