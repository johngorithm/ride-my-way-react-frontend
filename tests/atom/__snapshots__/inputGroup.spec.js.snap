// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests it does not render error message when error is undefined 1`] = `
<Fragment>
  <div>
    <p
      className="smaller"
    >
       
    </p>
    <InputField
      lable="Name"
    />
  </div>
</Fragment>
`;

exports[`Button Tests it matches snapshot 1`] = `
<Fragment>
  <div>
    <p
      className="smaller"
    >
       
      <span
        className="error-message"
      >
        this field is required
      </span>
    </p>
    <InputField
      lable="Password"
      name="password"
      onChange={[MockFunction]}
      type="text"
      value="fd89ya8sdf"
    />
  </div>
</Fragment>
`;
